### PR TITLE
:memo: add documentation for conversation errors

### PR DESCRIPTION
**Pull request by akirilyuk**

### CODE REVIEW

File: pages/Documents/ConversationalAI/Third-PartyBots/bot-conversation-errors.md [hunk 1]
```diff
@@ -0,0 +1,383 @@
...
+This error indicates that there is a unknown error while processing the consumer event.
+#### Reasons
+Mostly implementation errors on our end.
+#### Solutions
+Please reach out to LivePerson Support and provide the exact error details. You can do this by clicking copy on the relevant error.
+
+### com.liveperson.bot-connectors-worker.error.aivendor.dialogflowv2.send-query
+
+#### Description
+Could not send query to DialogFlow.
```
Comment (fahimmahmoodmir) [inline pages/Documents/ConversationalAI/Third-PartyBots/bot-conversation-errors.md:129]: `Could not send a query to DialogFlow.`

File: pages/Documents/ConversationalAI/Third-PartyBots/bot-conversation-errors.md [hunk 2]
```diff
@@ -0,0 +1,383 @@
...
+#### Reasons
+Mostly issues with DialogFlow APIS.
+#### Solutions
+Double check you have enough credits left for your Dialogflow account.
```
Comment (fahimmahmoodmir) [inline pages/Documents/ConversationalAI/Third-PartyBots/bot-conversation-errors.md:133]: `Double-check`

File: pages/Documents/ConversationalAI/Third-PartyBots/bot-conversation-errors.md [hunk 3]
```diff
@@ -0,0 +1,383 @@
...
+Check for any interruption on Dialogflow side.
+
+### com.liveperson.bot-connectors-worker.error.aivendor.faas.invoke-lambda
+
+#### Description
+This indicates an issue during Faas invocation.
+#### Reasons
+Mostly this happens if your quota limit is exceeded.
+#### Solutions
+Double check you have enough api calls left for Faas.
```
Comment (fahimmahmoodmir) [inline pages/Documents/ConversationalAI/Third-PartyBots/bot-conversation-errors.md:143]: `Double-check that you have enough API calls left for Faas.`

File: pages/Documents/ConversationalAI/Third-PartyBots/bot-conversation-errors.md [hunk 4]
```diff
@@ -0,0 +1,383 @@
...
+This happens if there is no response body provided.
+#### Reasons
+The Faas callback was invoked without a response body object.
+#### Solutions
+Provide response body in callback.
+
+### com.liveperson.bot-connectors-worker.error.aivendor.faas.failed-hook
+
+#### Description
+Indicates that the faas hook could not be invoked.
```
Comment (fahimmahmoodmir) [inline pages/Documents/ConversationalAI/Third-PartyBots/bot-conversation-errors.md:157]: `Faas `

File: pages/Documents/ConversationalAI/Third-PartyBots/bot-conversation-errors.md [hunk 5]
```diff
@@ -0,0 +1,383 @@
...
+#### Reasons
+Implementation errors inside Faas hook.
+#### Solutions
+Double check and test the Faas Hook code.
```
Comment (fahimmahmoodmir) [inline pages/Documents/ConversationalAI/Third-PartyBots/bot-conversation-errors.md:161]: `Double-check`

File: pages/Documents/ConversationalAI/Third-PartyBots/bot-conversation-errors.md [hunk 6]
```diff
@@ -0,0 +1,383 @@
...
+
+### com.liveperson.bot-connectors-worker.error.aivendor.lex.composite-format-not-supported
+
+#### Description
+This error is thrown Lex responds with a composite format.
+#### Reasons
+Lex response contained composite format.
```
Comment (fahimmahmoodmir) [inline pages/Documents/ConversationalAI/Third-PartyBots/bot-conversation-errors.md:168]: `a composite`

File: pages/Documents/ConversationalAI/Third-PartyBots/bot-conversation-errors.md [hunk 7]
```diff
@@ -0,0 +1,383 @@
...
+#### Solutions
+Do not use composite format in Lex for TPB.
```
Comment (fahimmahmoodmir) [inline pages/Documents/ConversationalAI/Third-PartyBots/bot-conversation-errors.md:170]: `Do not use the composite format in Lex for TPB.`

File: pages/Documents/ConversationalAI/Third-PartyBots/bot-conversation-errors.md [hunk 8]
```diff
@@ -0,0 +1,383 @@
...
+This indicates an issue during parsing.
+#### Reasons
+Usually malformed response from Lex.
+#### Solutions
+IF this issue happens frequently, please reach out to LP Support with the copied error details.
+
+### com.liveperson.bot-connectors-worker.error.aivendor.lex.send-query
+
+#### Description
+Happens if the query to Lex could not be send.
```
Comment (fahimmahmoodmir) [inline pages/Documents/ConversationalAI/Third-PartyBots/bot-conversation-errors.md:220]: `be sent.`

File: pages/Documents/ConversationalAI/Third-PartyBots/bot-conversation-errors.md [hunk 9]
```diff
@@ -0,0 +1,383 @@
...
+#### Reasons
+Lex temporarily unavailable.
+#### Solutions
+Double check the credentials for your bot are still valid and you have enough credits left
```
Comment (fahimmahmoodmir) [inline pages/Documents/ConversationalAI/Third-PartyBots/bot-conversation-errors.md:224]: `Double-check`

File: pages/Documents/ConversationalAI/Third-PartyBots/bot-conversation-errors.md [hunk 10]
```diff
@@ -0,0 +1,383 @@
...
+#### Description
+This error happens because the bot received a list response from watson, without any options inside.
+#### Reasons
+The bot doesn't know how to handle a empty list, therefore we assume this is an error.
+#### Solutions
+Provide options properties to your Watson List object.
+
+### com.liveperson.bot-connectors-worker.error.aivendor.watson.message-format-not-supported
+#### Description
+The message format provided by watson is not supported by our connector.
```
Comment (fahimmahmoodmir) [inline pages/Documents/ConversationalAI/Third-PartyBots/bot-conversation-errors.md:256]: `by Watson`

File: pages/Documents/ConversationalAI/Third-PartyBots/bot-conversation-errors.md [hunk 11]
```diff
@@ -0,0 +1,383 @@
...
+Double check the pause time property is provided
+
+### com.liveperson.bot-connectors-worker.error.aivendor.watson.parse-message.get-parse-message-failed
+
+#### Description
+This indicates an issue during message parsing.
+#### Reasons
+Wrong structure in the JSON payload.
+#### Solutions
+Double check the Watson responses here: [Watson Assistant](third-party-bots-ibm-watson-assistant.html)[Watson AssistantV2](third-party-bots-ibm-watson-assistant-version-2.html)
```
Comment (fahimmahmoodmir) [inline pages/Documents/ConversationalAI/Third-PartyBots/bot-conversation-errors.md:287]: `Double-check`

File: pages/Documents/ConversationalAI/Third-PartyBots/bot-conversation-errors.md [hunk 12]
```diff
@@ -0,0 +1,383 @@
...
+
+### com.liveperson.bot-connectors-worker.error.aivendor.watson.index.generic-parser-crashed
+
+#### Description
+This happens if there is an issue with parsing the provided watson response.
+#### Reasons
+Malformed Watson response.
+#### Solutions
+Double check the Watson responses here: [Watson Assistant](third-party-bots-ibm-watson-assistant.html)[Watson AssistantV2](third-party-bots-ibm-watson-assistant-version-2.html)
```
Comment (fahimmahmoodmir) [inline pages/Documents/ConversationalAI/Third-PartyBots/bot-conversation-errors.md:296]: `Double-check`

File: pages/Documents/ConversationalAI/Third-PartyBots/bot-conversation-errors.md [hunk 13]
```diff
@@ -0,0 +1,383 @@
...
+
+### com.liveperson.bot-connectors-worker.error.aivendor.watson.send-query
+
+#### Description
+This happens if we could not send a query to Watson. 
+#### Reasons
+Usually this happens if the quota limit is reached.
```
Comment (fahimmahmoodmir) [inline pages/Documents/ConversationalAI/Third-PartyBots/bot-conversation-errors.md:303]: `Usually, `

File: pages/Documents/ConversationalAI/Third-PartyBots/bot-conversation-errors.md [hunk 14]
```diff
@@ -0,0 +1,383 @@
...
+This could have multiple reasons. Mostly is a wrong implementation on Watson side.
+#### Solutions
+Check the error logs on your Watson Instance to find out the root cause of the failing bot.
+
+### com.liveperson.bot-connectors-worker.error.aivendor.watsonv2.send-query
+
+#### Description
+This happens if we could not send a query to Watson. 
+#### Reasons
+Usually this happens if the quota limit is reached.
```
Comment (fahimmahmoodmir) [inline pages/Documents/ConversationalAI/Third-PartyBots/bot-conversation-errors.md:321]: `Usually, `

File: pages/Documents/ConversationalAI/Third-PartyBots/bot-conversation-errors.md [hunk 15]
```diff
@@ -0,0 +1,383 @@
...
+Mostly, this would only fail because of a network issue.
+#### Solutions
+None. If this issue happens frequently, please reach out to LP Support with the error details.
+
+### com.liveperson.bot-connectors-worker.error.aivendor.watsonv2.invalid-session
+
+#### Description
+The session id used by the conversation is not valid anymore.
+#### Reasons
+The lifespan of a conversation state is different in LP and IBM. It could be that a open conversation in LP is already closed in Watson.
```
Comment (fahimmahmoodmir) [inline pages/Documents/ConversationalAI/Third-PartyBots/bot-conversation-errors.md:339]: `an open`

File: pages/Documents/ConversationalAI/Third-PartyBots/bot-conversation-errors.md [hunk 16]
```diff
@@ -0,0 +1,383 @@
...
+#### Solutions
+Increase the lifetime of a Watson conversation
+Alternatively, you can also auto close conversations from LP side remove any stuck conversations.
```
Comment (fahimmahmoodmir) [inline pages/Documents/ConversationalAI/Third-PartyBots/bot-conversation-errors.md:342]: `from The LP`

File: pages/Documents/ConversationalAI/Third-PartyBots/bot-conversation-errors.md [hunk 17]
```diff
@@ -0,0 +1,383 @@
...
+
+### com.liveperson.bot-connectors-worker.error.aivendor.microsoft.send-query
+
+#### Description
+Sending query to MS Bot Framework failed.
+#### Reasons
+This usually happens if the Endpoint is not respoding in time.
```
Comment (fahimmahmoodmir) [inline pages/Documents/ConversationalAI/Third-PartyBots/bot-conversation-errors.md:349]: `This usually happens if the Endpoint is not responding in time.`

File: pages/Documents/ConversationalAI/Third-PartyBots/bot-conversation-errors.md [hunk 18]
```diff
@@ -0,0 +1,383 @@
...
+#### Solutions
+Make sure you responde in time.
```
Comment (fahimmahmoodmir) [inline pages/Documents/ConversationalAI/Third-PartyBots/bot-conversation-errors.md:351]: `Make sure you send the response in time.`

File: pages/Documents/ConversationalAI/Third-PartyBots/bot-conversation-errors.md [hunk 19]
```diff
@@ -0,0 +1,383 @@
...
+If you are using a custom endpoint, double check it behaves exactly as the one we expect [Microsoft Bots](third-party-bots-microsoft-bot-framework.html)
+### com.liveperson.bot-connectors-worker.error.aivendor.microsoft.parser-crashed
+
+#### Description
+The parser could not parse the provided input and therefore crashed.
+#### Reasons
+Malformed JSON, malformed payload.
+#### Solutions
+Double check the payload and make sure it is same as described [here](third-party-bots-microsoft-bot-framework.html)
```
Comment (fahimmahmoodmir) [inline pages/Documents/ConversationalAI/Third-PartyBots/bot-conversation-errors.md:361]: `Double-check`

File: pages/Documents/ConversationalAI/Third-PartyBots/bot-conversation-errors.md [hunk 20]
```diff
@@ -0,0 +1,383 @@
...
+Double check the payload and make sure it is same as described [here](third-party-bots-microsoft-bot-framework.html) 
+
+### com.liveperson.bot-connectors-worker.error.aivendor.microsoft.response-missing
+
+#### Description
+There is no response provided by the MS bot.
+#### Reasons
+Implementation error on MS bot.
+#### Solutions
+Double check the payload and make sure it is same as described [here](third-party-bots-microsoft-bot-framework.html)
```
Comment (fahimmahmoodmir) [inline pages/Documents/ConversationalAI/Third-PartyBots/bot-conversation-errors.md:370]: `Double-Check`

File: pages/Documents/ConversationalAI/Third-PartyBots/bot-conversation-errors.md [hunk 21]
```diff
@@ -0,0 +1,383 @@
...
+Double check that you are only using supported types as described [here](third-party-bots-microsoft-bot-framework.html) 
+
+### com.liveperson.bot-connectors-worker.error.aivendor.microsoft.activity.invalid-name
+
+#### Description
+The provided activity name is not supported.
+#### Reasons
+The bot is not capable of handling this activity name.
+#### Solutions
+Double check that you are only using supported activity names as described [here](third-party-bots-microsoft-bot-framework.html)
```
Comment (fahimmahmoodmir) [inline pages/Documents/ConversationalAI/Third-PartyBots/bot-conversation-errors.md:388]: `Double-check`

**Review by fahimmahmoodmir — CHANGES_REQUESTED** [review]

**Comment by lmartliveperson** [comment]
Merged. Please verify post merge.

**Pull request merged**